.gitlab-ci.yml

```diff
@@ -781,79 +854,54 @@ publish-rustdoc:
   script:
-    # If $CI_COMMIT_REF_NAME doesn't match one of $RUSTDOCS_DEPLOY_REFS space-separated values, we
-    # exit immediately.
-    # Putting spaces at the front and back to ensure we are not matching just any substring, but the
-    # whole space-separated value.
-    - '[[ " ${RUSTDOCS_DEPLOY_REFS} " =~ " ${CI_COMMIT_REF_NAME} " ]] || exit 0'
+    - *test-build-rustdoc-script
     - rm -rf /tmp/*
-    # Set git config
-    - rm -rf .git/config
-    - git config user.email "devops-team@parity.io"
-    - git config user.name "${GITHUB_USER}"
-    - git config remote.origin.url "https://${GITHUB_TOKEN}@github.com/paritytech/${CI_PROJECT_NAME}.git"
-    - git config remote.origin.fetch "+refs/heads/*:refs/remotes/origin/*"
-    - git fetch origin gh-pages
-    # Install `ejs` and generate index.html based on RUSTDOCS_DEPLOY_REFS
-    - yarn global add ejs
-    - 'ejs ${INDEX_TPL} -i "{\"deploy_refs\":\"${RUSTDOCS_DEPLOY_REFS}\",\"repo_name\":\"${CI_PROJECT_NAME}\",\"latest\":\"${LATEST}\"}" > /tmp/index.html'
-    # Save README and docs
-    - cp -r ./crate-docs/ /tmp/doc/
-    - cp README.md /tmp/doc/
-    - git checkout gh-pages
-    # Remove directories no longer necessary, as specified in $RUSTDOCS_DEPLOY_REFS.
-    # Also ensure $RUSTDOCS_DEPLOY_REFS is not just empty spaces.
-    # Even though this block spans multiple lines, they are concatenated to run as a single line
-    #   command, so note for the semi-colons in the inner-most code block.
-    - if [[ ! -z ${RUSTDOCS_DEPLOY_REFS// } ]]; then
-        for FILE in *; do
-          if [[ ! " $RUSTDOCS_DEPLOY_REFS " =~ " $FILE " ]]; then
-            echo "Removing ${FILE}...";
-            rm -rf $FILE;
-          fi
-        done
-      fi
-    # Move the index page & built back
+    - mv -f crate-docs/${BUILD_REF} /tmp/docs
+    - *reset-gh-info-script
+    - git checkout -f gh-pages
+    # Upsert <li/> entry into the index.html
+    - cp index.html /tmp
+    - *upsert-index-script
     - mv -f /tmp/index.html .
-    # Ensure the destination dir doesn't exist.
-    - rm -rf ${CI_COMMIT_REF_NAME}
-    - mv -f /tmp/doc ${CI_COMMIT_REF_NAME}
+    # Ensure the destination dir is clear
+    - rm -rf ${BUILD_REF}
+    - mv -f /tmp/docs ${BUILD_REF}
     # Add the symlink
-    - '[[ -e "$LATEST" ]] && ln -sf "${LATEST}" latest'
-    # Upload files
-    - git add --all --force
-    # `git commit` has an exit code of > 0 if there is nothing to commit.
-    # This causes GitLab to exit immediately and marks this job failed.
-    # We don't want to mark the entire job failed if there's nothing to
-    # publish though, hence the `|| true`.
-    - git commit -m "___Updated docs for ${CI_COMMIT_REF_NAME}___" ||
-        echo "___Nothing to commit___"
-    - git push origin gh-pages --force
-  after_script:
-    - rm -rf .git/ ./*
+    - '[[ -e "$LATEST" ]] && { [[ -h latest ]] && rm latest; ln -sf "${LATEST}" latest; }'
+    - *git-force-push-rm-script
+
+rm-rustdoc:
```

Comment:
it's an unnecessary complication to hold this job in the CI config, it's not automation when you have a job that's going to be triggered rarely manually.
It's better to describe the process in the doc. Or put the whole rustdoc update logic into a separate script and condition its' use with input variables. Say, check first for `$RM_RUSTDOC_REF` then remove and exit.